astr303a.py

```diff
@@ -0,0 +1,571 @@
+        radians = projecteddistance(dataarray[i][1],dataarray[i][2],comaRA,comaDEC,dcoma)
+        distance = dcoma*radians
+        dataarray[i].append(distance)
+        
+    return dataarray
+
+#Plot CDF 
+def part1213(dataarray):
+
+    #Plot galaxies as radius vs. luminosity
+    plt.title('Figure 5: Coma Cluster Member Galaxy Luminosities')
+    plt.xlim(0,1.5)
+    plt.yscale('log')
+    plt.xlabel(r'Radial Distance '+r'$R\,(Mpc)$')
+    plt.ylabel(r'Luminosity '+r'$L_{gal}\,(10^{10}L_\odot)$')
+    for l in range(0,len(dataarray)):
+        plt.plot(dataarray[l][11],dataarray[l][10]/(10**10),color='black',marker='o',markersize=3,fillstyle=None)
+    pp.savefig()
+    plt.clf()
+    
+    #Calculate total luminosity
+    totallum=0
+    for i in range(0,len(dataarray)):
+        totallum += dataarray[i][10]
+
+    #caculate cummulative luminosity distribution in increments
+    sumlumin = [] #store sum of luminosities at increments of R   
+    radials = np.arange(0,1.51,0.02) #Increment R 75 times from 0-1.5Mpc
+    #Compute CDF 
+    for i in range(0,len(radials)):
+        lumin=0
+        for j in range(0,len(dataarray)):
+            if dataarray[j][11]<radials[i]:
+                lumin += dataarray[j][10]
+        sumlumin.append(lumin/totallum)
+
+    #Format plot
+    plt.title('Figure 6: Coma Cluster Cumulative Luminosity Distribution')
+    plt.xlim(0,1.5)
+    plt.ylim(0,1)
+    plt.xlabel(r'Radial Distance '+r'$R\,(Mpc)$')
+    plt.ylabel(r'Normalized Luminosity '+r'$L_{total}\,(3.94\ast10^{10}L_\odot)$')
+
+    #Define expontial function to fit to CDF
+    def func(x,a,b,c):
+        return a*x**b + c
+    #Fit CDF to func
+    cdf,pcov = optimize.curve_fit(func, radials, sumlumin, p0 = [0, 1, 0])
+    print('CDF fit [y=ax^b+c]: '+str(cdf))
+
+    #calculate Reff,r1/2
+    reff = ((0.5-cdf[2])/cdf[0])**(1/cdf[1])
+    rhalf = 4*reff/3
+    
+    print('Reff: '+str(reff)+' Mpc')
+    print('r1/2: '+str(rhalf)+' Mpc')
+    print('L total: '+str(totallum)+' solar luminosities')
+
+    #plot CDF, CDF fit. Compute and add Reff, r1/2, and L1/2 as short lines
+    x = np.arange(0,1.5,0.01)
+    plt.plot(x,func(x,*cdf),linestyle='--',color='black')
+    plt.plot(radials,sumlumin,color='black',marker=None,markersize=3,fillstyle=None)
+    plt.axvline(x=reff, ymin=0.4, ymax = .6, linewidth=1, color='black',linestyle=':')
+    plt.axvline(x=rhalf, ymin=func(rhalf,*cdf)-.09, ymax = func(rhalf,*cdf)+.1, linewidth=1, color='black',linestyle=':')
+    plt.axhline(y = func(rhalf,*cdf), xmin=rhalf-0.4, xmax=rhalf-0.21, linewidth=1, color='black',linestyle=':')
+    plt.annotate('$R_{eff}$',(reff+0.01,0.42))
+    plt.annotate('$r_{1/2}$',(rhalf+0.01,func(rhalf,*cdf)-.08))
+    plt.annotate('$L_{1/2}$',(rhalf+0.16,func(rhalf,*cdf)))
+    pp.savefig()
+    plt.clf()
+    
+    return cdf,reff,rhalf,totallum
+
+#Calculate recessional velocity for a redshift z
+def velocity(z):
+    c = 2.99792458E5
```

Comment:
Again. You've defined this constant elsewhere, and it's already defined in the utility libraries that you are using.